client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteSchemasAdmin.java

```diff
@@ -0,0 +1,218 @@
+package org.infinispan.client.hotrod;
+
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.schema.Schema;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * Interface for managing schema operations such as create, update, delete, and retrieve.
+ *
+ * @since 16.0
+ * @author Katia Aresti
+ */
+public interface RemoteSchemasAdmin {
+    /**
+     * Gets the schema by name.
+     *
+     * @param schemaName the name of the schema
+     * @return optional schema
+     */
+    Optional<Schema> get(String schemaName);
+
+    /**
+     * Non-blocking version of {@link RemoteSchemasAdmin#get(String)}
+     */
+    CompletionStage<Optional<Schema>> getAsync(String schemaName);
+
+    /**
+     * Gets the schema error by name, if such exists.
```

Comment:
It is probably worth to extend a bit more by saying if the schema has errors (parsing error or incompatibilities), this method should be invoked to retrieve them. 
Also, the create/update should recommend users to invoke this after they create/update a schema, to ensure it is correct.